### PR TITLE
Учебный проект: холст

### PR DESCRIPTION
**Pull request by elismountain**
I have changed the direction of the Histogram, added text on top of the histogram. 

---
:mortar_board: [Учебный проект: холст](https://up.htmlacademy.ru/javascript/21/user/89110/tasks/4)

:boom: https://htmlacademy-javascript.github.io/89110-code-and-magick-21/2/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,11 +11,19 @@
     "url": "https://github.com/htmlacademy-javascript/89110-code-and-magick-21/issues"
   },
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^4.2.0",
```
Comment (cy6erskunk) [inline package.json:14]: These seem to be unrelated changes, most probably they are result of applying some some tutorial to the project.

File: index.html [hunk 2]
```diff
@@ -180,6 +180,8 @@ <h4 class="setup-similar-title">Похожие персонажи</h4>
   </div>
 </template>
 
+<canvas id="canvas" width="1000" style="width: 100%;"></canvas>
```
Comment (cy6erskunk) [inline index.html:183]: This `canvas` is not need as the game canvas is created when the game is initialized.

File: js/stat.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    ctx.fillText(
+        Math.floor(times[i]),
+        CLOUD_X + HIST_GAP + (BAR_GAP + BAR_WIDTH) * i,
+        CLOUD_Y + CLOUD_HEIGHT - HIST_GAP - (HIST_HEIGHT * times[i]) / maxTime - FONT_GAP
+    );
+  }
+
+  for (i = 0; i < players.length; i++) {
+    if (i === 0) {
```
Comment (cy6erskunk) [inline js/stat.js:74]: > Цвет колонки игрока **Вы** `rgba(255, 0, 0, 1)`

But it does not mean that **Вы** is always the first element in the array of players

**Comment by cy6erskunk** [comment]
The functions `getRandomOpacity`, `renderCloud`, and `getMaxElement` are really good. But the `window.renderStatistics` is still very bloated: a separate function to render a histogram bar and maybe a function to render the whole histogram (which uses the function to render a bar) would help with readability a lot; also, extracting function to set up text properties and then render it could be helpful fo reading.

File: js/stat.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+var GAP = 10;
+var FONT_GAP = 15;
+
+var HIST_HEIGHT = 150;
+var BAR_WIDTH = 40;
+var BAR_GAP = 50;
+var HIST_GAP = 40;
+
+var getRandomOpacity = function (red, green, blue) {
+  return 'rgba(' + red + ', ' + green + ', ' + blue + ', ' + Math.random() + ')';
```
Comment (cy6erskunk) [inline js/stat.js:16]: > Цвет колонок других игроков — синий, а насыщенность задаётся случайным образом

The last parameter in `rgba` is opacity, which is not the same as `saturation` in the `hsl` color model.

File: package.json (hunk 1 above)
Comment (elismountain) [reply package.json:14]: Should I just delete them then?

Comment (cy6erskunk) [reply package.json]: Yes, I would just get rid of these changes.

File: js/stat.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+var HIST_GAP = 40;
+
+var getRandomInt = function (min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min) + min);
+};
+
+var getRandomSaturation = function (hue, lightness) {
+  return 'hsl(' + hue + ', ' + getRandomInt(0, 100) + '%, ' + lightness + '%)';
```
Comment (cy6erskunk) [inline js/stat.js:22]: It seems the function returns a color with saturation values between 0 to 99 (not 100) as `Math.random` returns a floating-point, pseudo-random number between 0 (inclusive) and 1 (exclusive).

**Review by cy6erskunk — CHANGES_REQUESTED** [review]
Looks really good now, just a couple of small fixes:
- make `getRandomSaturation` return colors with saturation including 100%
- remove package.json changes (`npm install` should remove the changes from `package-lock.json`)

**Review by cy6erskunk — CHANGES_REQUESTED** [review]
Let's revert changes in `package.json` and `package-lock.json` and that's it!

**Pull request merged**